pyre/extra_tests/snippets/stdlib_posix.py

```
@@ -33,3 +38,197 @@ def __int__(self):
+
+    def values(self):
+        return self._values
+
+    def __getitem__(self, key):
+        raise AssertionError("__getitem__ must not be called")
+
+
+_MISSING = "/nonexistent-program-" + str(os.getpid())
+
+for _short in (_Env([b"A"], [b"1"], 3), _Env([b"A", b"B"], [b"1"], 2)):
+    assert_raises(IndexError, lambda e=_short: os.execve(_MISSING, ["x"], e))
```

Ruff (0.16.1)

[error] Starting a process without a shell

(S606)